### ISSUE TITLE
Fix MIECD Vietnam Log Face Bug

### ISSUE THREAD

**Issue by moshthepitt** [OpenSRP/opensrp-web#235]
When viewing the data on any one of the "Log Face" pages, a request is sent to fetch data from Superset.  If Superset returns no data then the page will not return any message to the user but the Loading spinner will keep on spinning (probably for eternity).

![OpenSRP](https://user-images.githubusercontent.com/372073/78116093-7899db00-740c-11ea-968b-a3b0eb6aa14f.png)

**Comment by moshthepitt** [comment]
This can be fixed by checking that `supersetFetch` returned something and if not removing the loading icon and showing some kind of message e.g. "no records found"

**Comment by cKellyDesign** [comment]
> This can be fixed by checking that supersetFetch returned something

@moshthepitt Should this be something that's baked into the connector?

If it was then the connector could reject the promise or emit an error to be caught by the client's service, which is then passed to the FE container/component (who is utilizing the service) to display some sort of error message / apply some fallback business logic.

**Comment by moshthepitt** [comment]
@cKellyDesign yes I think we should bake it in.  But I think we may need to do that as part of [moving supersetFetch](https://github.com/onaio/js-tools/issues/132) into a package.

**Comment by cKellyDesign** [comment]
@moshthepitt is there an adjacent ticket in [js-tools](https://github.com/onaio/js-tools/issues)?

> But I think we may need to do that as part of [moving supersetFetch](https://github.com/onaio/js-tools/issues/132) into a package.

Agreed! Are you thinking this will be like a higher order wrapper for the SupersetConnector package or a service submodule of the package?

**Comment by moshthepitt** [comment]
@cKellyDesign 

> @moshthepitt is there an adjacent ticket in js-tools?

I think it should be done as part of https://github.com/onaio/js-tools/issues/132 - I left a comment to that effect.

> Are you thinking this will be like a higher order wrapper for the SupersetConnector package or a service submodule of the package?

I think it should be a service module for SupersetConnector
